samcli/commands/init/interactive_init_flow.py

```diff
@@ -40,29 +45,90 @@ def do_interactive(
+    no_input,
+    location_opt_choice,
+):
+    if location_opt_choice == "1":
+        _generate_simple_application(
+            location, pt_explicit, runtime, base_image, dependency_manager, output_dir, name, app_template, package_type
+        )
+
+    elif location_opt_choice == "2":
+        _generate_from_use_case(
+            location, pt_explicit, package_type, runtime, base_image, dependency_manager, output_dir, name, app_template
+        )
+
+    else:
         _generate_from_location(
             location, package_type, runtime, dependency_manager, output_dir, name, app_template, no_input
         )
-    else:
-        if not pt_explicit:
-            click.echo("What package type would you like to use?")
-            click.echo("\t1 - Zip (artifact is a zip uploaded to S3)\t")
-            click.echo("\t2 - Image (artifact is an image uploaded to an ECR image repository)")
-            package_opt_choice = click.prompt("Package type", type=click.Choice(["1", "2"]), show_choices=False)
-            if package_opt_choice == "1":
-                package_type = ZIP
-            else:
-                package_type = IMAGE
-
-        _generate_from_app_template(
-            location, package_type, runtime, base_image, dependency_manager, output_dir, name, app_template
+
+
+# pylint: disable=too-many-statements
+def _generate_simple_application(
+    location, pt_explicit, runtime, base_image, dependency_manager, output_dir, name, app_template, package_type=ZIP
+):
+    if pt_explicit and package_type == IMAGE:
+        raise InvalidInitOptionException("Hello World Application only supports ZIP package type")
+
+    templates = InitTemplates()
+    if not runtime:
+        question = (
+            "Which runtime would you like to use? "
+            + "We will default to the latest supported version of your selected runtime."
         )
+        runtime_list = sorted(RUNTIME_DEP_TEMPLATE_MAPPING.keys())
+        runtime_choosen = _get_choice_from_options(runtime, runtime_list, question, "Runtime")
+        runtime_templates = RUNTIME_DEP_TEMPLATE_MAPPING[runtime_choosen]
+        runtime = runtime_templates[0]["runtimes"][0]
+
+    dependency_manager = _get_dependency_manager(None, dependency_manager, runtime)
+    bundle_template = templates.get_bundle_option(package_type, runtime, dependency_manager)
+    template = bundle_template[0]
+    app_template = template["appTemplate"]
+    location = template["init_location"]
+
+    if not name:
+        name = click.prompt("\nProject name", type=str, default="sam-app")
+    extra_context = {"project_name": name, "runtime": runtime}
+
+    no_input = True
+    summary_msg = generate_summary_message(
+        package_type, runtime, base_image, dependency_manager, output_dir, name, app_template
+    )
+    click.echo(summary_msg)
+    do_generate(location, package_type, runtime, dependency_manager, output_dir, name, no_input, extra_context)
 
 
 def _generate_from_location(
```

Comment:
We could also use the `generate_summary_message` here.
Add a `location=None` parameter at the end and return this message if it's set.

Reply:
`location` would always be set before calling `do_generate` and location is part of the message return.
The only way to add this `generate_summary_message` is add a flag, which i truly think shouldn't be used and all the number of argument for the `generate_summary_message` is already too much.